include/miral/miral/bounce_keys.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright © Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 or 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef MIRAL_BOUNCE_KEYS_H
+#define MIRAL_BOUNCE_KEYS_H
+
+#include <chrono>
+#include <functional>
+#include <memory>
+
+namespace mir { class Server; }
+
+namespace miral
+{
+/// Enables configuring bounce keys at runtime.
+/// \remark Since MirAL 5.4
+class BounceKeys
+{
+public:
+    /// Creates a `BounceKeys` instance that's enabled by default.
+    static auto enabled() -> BounceKeys;
+
+    /// Creates a `BounceKeys` instance that's disabled by default.
+    static auto disabled() -> BounceKeys;
```

Comment:
It would be good to provide integration with live configuration:

```suggestion
    /// Creates a `BounceKeys` instance that's enabled by default.
    static auto enabled() -> BounceKeys;

    /// Creates a `BounceKeys` instance that's disabled by default.
    static auto disabled() -> BounceKeys;
    
    /// Creates a `BounceKeys` instance that's configured by a live config store
    explicit BounceKeys(live_config::Store& config_store);

```